*** The reference

--- Yours

***************

*** 1,28 ****

! #
! _program
!   #
!   _class
!     Test
!     Object
!     "tests/good3.test"
!     (
!     #
!     _attr
!       foo
!       Int
!       #
!       _no_expr
!       : _no_type
!     #
!     _method
!       bar
!       Object
!       #
!       _assign
!         foo
!         #
!         _int
!           3
!         : _no_type
!       : _no_type
!     )
--- 1 ----

! "tests/good3.test", line 2: parse error at or near OBJECTID = foo
