*** The reference

--- Yours

***************

*** 13,30 ****

        #
        _plus
          #
          _plus
-           #
-           _static_dispatch
-             #
-             _object
-               a
-             : _no_type
-             Bazz
-             doh
-             (
-             )
-           : _no_type
            #
            _dispatch
              #
--- 13,30 ----

        #
        _plus
          #
+         _static_dispatch
+           #
+           _object
+             a
+           : _no_type
+           Bazz
+           doh
+           (
+           )
+         : _no_type
+         #
          _plus
            #
            _dispatch
              #
***************

*** 35,50 ****

              (
              )
            : _no_type
-         : _no_type
-         #
-         _dispatch
            #
!           _object
!             e
            : _no_type
-           doh
-           (
-           )
          : _no_type
        : _no_type
      )
--- 35,50 ----

              (
              )
            : _no_type
            #
!           _dispatch
!             #
!             _object
!               e
!             : _no_type
!             doh
!             (
!             )
            : _no_type
          : _no_type
        : _no_type
      )
