*** The reference

--- Yours

***************

*** 1,33 ****

! #
! _program
!   #
!   _class
!     Test
!     Object
!     "tests/good2.test"
!     (
!     #
!     _attr
!       foo
!       Test
!       #
!       _no_expr
!       : _no_type
!     #
!     _method
!       baz
!       String
!       #
!       _let
!         x
!         Int
!         #
!         _int
!           5
!         : _no_type
!         #
!         _object
!           x
!         : _no_type
!       : _no_type
!     )
--- 1 ----

! "tests/good2.test", line 2: parse error at or near OBJECTID = foo
