*** The reference

--- Yours

***************

*** 662,706 ****

                    _assign
                      int
                      #
!                     _plus
!                       #
!                       _mul
!                         #
!                         _object
!                           int
!                         : _no_type
                          #
                          _int
                            10
                          : _no_type
-                       : _no_type
-                       #
-                       _dispatch
-                         #
-                         _object
-                           self
-                         : _no_type
-                         c2i
-                         (
                          #
                          _dispatch
                            #
                            _object
!                             s
!                           : _no_type
!                           substr
                            (
                            #
!                           _object
!                             i
!                           : _no_type
!                           #
!                           _int
!                             1
                            : _no_type
                            )
                          : _no_type
-                         )
                        : _no_type
                      : _no_type
                    : _no_type
--- 662,706 ----

                    _assign
                      int
                      #
!                     _mul
!                       #
!                       _object
!                         int
!                       : _no_type
!                       #
!                       _plus
                          #
                          _int
                            10
                          : _no_type
                          #
                          _dispatch
                            #
                            _object
!                             self
!                           : _no_type
!                           c2i
                            (
                            #
!                           _dispatch
!                             #
!                             _object
!                               s
!                             : _no_type
!                             substr
!                             (
!                             #
!                             _object
!                               i
!                             : _no_type
!                             #
!                             _int
!                               1
!                             : _no_type
!                             )
                            : _no_type
                            )
                          : _no_type
                        : _no_type
                      : _no_type
                    : _no_type
