*** The reference

--- Yours

***************

*** 1,2 ****

! tests/bad2.test:3: Formal parameter x is multiply defined.
! Compilation halted due to static semantic errors.
--- 1,33 ----

! #
! _program
!   #
!   _class
!     Main
!     Object
!     "tests/bad2.test"
!     (
!     #
!     _method
!       main
!       Int
!       #
!       _int
!         1
!       : _no_type
!     #
!     _method
!       foo
!       #
!       _formal
!         x
!         Int
!       #
!       _formal
!         x
!         Bool
!       Object
!       #
!       _new
!         Object
!       : _no_type
!     )
