*** The reference

--- Yours

***************

*** 1,2 ****

! tests/bad5.test:4: Formal parameter x cannot have type TreeConstants.SELF_TYPE.
! Compilation halted due to static semantic errors.
--- 1,36 ----

! #
! _program
!   #
!   _class
!     Main
!     Object
!     "tests/bad5.test"
!     (
!     #
!     _method
!       main
!       Int
!       #
!       _int
!         0
!       : _no_type
!     )
!   #
!   _class
!     A
!     Object
!     "tests/bad5.test"
!     (
!     #
!     _method
!       foo
!       #
!       _formal
!         x
!         SELF_TYPE
!       Object
!       #
!       _object
!         x
!       : _no_type
!     )
