*** The reference

--- Yours

***************

*** 1,108 ****

! #
! _program
!   #
!   _class
!     A
!     Object
!     "tests/good3.test"
!     (
!     #
!     _method
!       foo
!       #
!       _formal
!         a
!         Int
!       #
!       _formal
!         b
!         B
!       #
!       _formal
!         c
!         A
!       #
!       _formal
!         d
!         B
!       SELF_TYPE
!       #
!       _object
!         self
!       : SELF_TYPE
!     )
!   #
!   _class
!     B
!     A
!     "tests/good3.test"
!     (
!     #
!     _method
!       moo
!       SELF_TYPE
!       #
!       _let
!         b
!         C
!         #
!         _new
!           C
!         : C
!         #
!         _dispatch
!           #
!           _object
!             self
!           : SELF_TYPE
!           foo
!           (
!           #
!           _int
!             4
!           : Int
!           #
!           _object
!             b
!           : C
!           #
!           _object
!             b
!           : C
!           #
!           _object
!             b
!           : C
!           )
!         : SELF_TYPE
!       : SELF_TYPE
!     )
!   #
!   _class
!     C
!     B
!     "tests/good3.test"
!     (
!     #
!     _attr
!       foo
!       Int
!       #
!       _no_expr
!       : _no_type
!     )
!   #
!   _class
!     Main
!     Object
!     "tests/good3.test"
!     (
!     #
!     _method
!       main
!       Int
!       #
!       _int
!         6
!       : Int
!     )
--- 1,4 ----

! tests/good3.test:4: Undeclared identifier self.
! tests/good3.test:4: Inferred return type Object of method foo does not conform to declared return type SELF_TYPE.
! tests/good3.test:13: Inferred return type Object of method moo does not conform to declared return type SELF_TYPE.
! Compilation halted due to static semantic errors.
