*** The reference

--- Yours

***************

*** 1,110 ****

! #
! _program
!   #
!   _class
!     Count
!     Object
!     "tests/good5.test"
!     (
!     #
!     _attr
!       i
!       Int
!       #
!       _int
!         0
!       : Int
!     #
!     _method
!       inc
!       SELF_TYPE
!       #
!       _block
!         #
!         _assign
!           i
!           #
!           _plus
!             #
!             _object
!               i
!             : Int
!             #
!             _int
!               1
!             : Int
!           : Int
!         : Int
!         #
!         _object
!           self
!         : SELF_TYPE
!       : SELF_TYPE
!     )
!   #
!   _class
!     Stock
!     Count
!     "tests/good5.test"
!     (
!     #
!     _attr
!       name
!       String
!       #
!       _no_expr
!       : _no_type
!     #
!     _method
!       get
!       String
!       #
!       _object
!         name
!       : String
!     )
!   #
!   _class
!     Main
!     Object
!     "tests/good5.test"
!     (
!     #
!     _attr
!       a
!       Stock
!       #
!       _no_expr
!       : _no_type
!     #
!     _method
!       main
!       Object
!       #
!       _block
!         #
!         _assign
!           a
!           #
!           _dispatch
!             #
!             _new
!               Stock
!             : Stock
!             inc
!             (
!             )
!           : Stock
!         : Stock
!         #
!         _dispatch
!           #
!           _object
!             a
!           : Stock
!           get
!           (
!           )
!         : String
!       : String
!     )
--- 1,2 ----

! tests/good5.test:8: Inferred return type Object of method inc does not conform to declared return type SELF_TYPE.
! Compilation halted due to static semantic errors.
