*** The reference

--- Yours

***************

*** 17,56 ****

            #
            _int
              5
!           : Int
            copy
            (
            )
!         : Int
          #
          _dispatch
            #
            _string
              "test"
!           : String
            copy
            (
            )
!         : String
          #
          _dispatch
            #
            _bool
              1
!           : Bool
            copy
            (
            )
!         : Bool
          #
          _dispatch
            #
            _string
              "test"
!           : String
            length
            (
            )
!         : Int
!       : Int
      )
--- 17,56 ----

            #
            _int
              5
!           : Object
            copy
            (
            )
!         : Object
          #
          _dispatch
            #
            _string
              "test"
!           : Object
            copy
            (
            )
!         : Object
          #
          _dispatch
            #
            _bool
              1
!           : Object
            copy
            (
            )
!         : Object
          #
          _dispatch
            #
            _string
              "test"
!           : Object
            length
            (
            )
!         : Object
!       : Object
      )
